arbnode/node.go

```diff
@@ -563,7 +565,25 @@ func createNodeImpl(
 	if err != nil {
 		return nil, err
 	}
-	inboxReader, err := NewInboxReader(inboxTracker, l1client, l1Reader, new(big.Int).SetUint64(deployInfo.DeployedAt), delayedBridge, sequencerInbox, func() *InboxReaderConfig { return &configFetcher.Get().InboxReader })
+	firstMessageBlock := new(big.Int).SetUint64(deployInfo.DeployedAt)
+	if config.SnapSyncTest.Enabled {
+		firstMessageToRead := config.SnapSyncTest.DelayedCount
+		if firstMessageToRead > config.SnapSyncTest.BatchCount {
+			firstMessageToRead = config.SnapSyncTest.BatchCount
+		}
+		if firstMessageToRead > 0 {
+			firstMessageToRead--
+		}
+		// Find the first block containing the first message to read
+		// Subtract 1 to get the block before the first message to read,
+		// this is done to fetch previous batch metadata needed for snap sync.
+		block, err := FindBlockContainingBatch(ctx, deployInfo.Rollup, l1client, config.SnapSyncTest.ParentChainAssertionBlock, firstMessageToRead-1)
```

Comment:
This is called FindBlockContainingBatch, but seems to actually find block containing node(assertion).
What's actually needed is to find the block containing the first delayed message that's included in the batch (or the batch itself if it has no new delayedInbox messages)

Reply:
Done.